tests/ImageSharp.Tests/TestUtilities/Attributes/RuntimeFeatures/IntrinsicTestsUtils.cs

```diff
@@ -0,0 +1,373 @@
+
+            // AVX
+            UncheckIfSupported(ref notSupported, RuntimeFeature.Vector8, HasVector8);
+            UncheckIfSupported(ref notSupported, RuntimeFeature.AVX, HasAvx);
+            UncheckIfSupported(ref notSupported, RuntimeFeature.AVX2, HasAvx2);
+
+            // BMI
+            UncheckIfSupported(ref notSupported, RuntimeFeature.BMI1, HasBmi1);
+            UncheckIfSupported(ref notSupported, RuntimeFeature.BMI2, HasBmi2);
+
+            // ABM
+            UncheckIfSupported(ref notSupported, RuntimeFeature.POPCNT, HasPOPCNT);
+            UncheckIfSupported(ref notSupported, RuntimeFeature.LZCNT, HasLZCNT);
+
+            // FMA
+            UncheckIfSupported(ref notSupported, RuntimeFeature.FMA, HasFma);
+
+            // AES
+            UncheckIfSupported(ref notSupported, RuntimeFeature.AES, HasAes);
+
+            // PCLMULQDQ
+            UncheckIfSupported(ref notSupported, RuntimeFeature.PCLMULQDQ, HasPCLMULQDQ);
+
+            return notSupported;
+        }
+
+        private static void UncheckIfSupported(ref RuntimeFeature features, RuntimeFeature value, bool isSupported)
+        {
+            if (isSupported)
+            {
+                features &= ~value;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Flags for determining if testing environment supports certain runtime features
+    /// </summary>
+    [Flags]
+    public enum RuntimeFeature
+    {
+        /// <summary>
+        /// No flags set
+        /// </summary>
+        /// <remarks>Used internally, general use case should not use this option.</remarks>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        None = 0,
+
+        /// <summary>
+        /// SSE hardware support
+        /// </summary>
+        /// <remarks>Should be used only for testing <see cref="Vector"/> based intrinsic code</remarks>
+        Vector4 = 1 << 0,
+
+        /// <summary>
+        /// Equivalent of Sse.IsSupported
+        /// </summary>
+        /// <remarks>Not supported in certain runtimes even if hardware supports it</remarks>
+        SSE = 1 << 1,
+
+        /// <summary>
+        /// Equivalent of Sse2.IsSupported
+        /// </summary>
+        /// <remarks>Not supported in certain runtimes even if hardware supports it</remarks>
+        SSE2 = 1 << 2,
+
+        /// <summary>
+        /// Equivalent of Sse3.IsSupported
+        /// </summary>
+        /// <remarks>Not supported in certain runtimes even if hardware supports it</remarks>
+        SSE3 = 1 << 3,
+
+        /// <summary>
+        /// Equivalent of Ssse3.IsSupported
+        /// </summary>
+        /// <remarks>Not supported in certain runtimes even if hardware supports it</remarks>
+        SSSE3 = 1 << 4,
+
+        /// <summary>
+        /// Equivalent of Sse41.IsSupported
+        /// </summary>
+        /// <remarks>Not supported in certain runtimes even if hardware supports it</remarks>
+        SSE41 = 1 << 5,
```

Comment:
[Naming] The BCL naming conventions for enums are the same as for method or class names. These should be `Sse`, `Sse2`, `Sse41`, `Popcnt` etc. The test project's analyzers may be more relaxed on this rule, but in this case I don't see a reason to not conform them.

Reply:
Done!